src/components/updatemenus/attributes.js

```diff
@@ -49,6 +49,7 @@ var buttonsAttrs = {
 
 module.exports = {
     _isLinkedToArray: 'updatemenu',
+    _arrayAttrRegexps: [/^updatemenus\[(0|[1-9][0-9]+)\]\.buttons/],
```

Comment:
I'm ok with adding `_arrayAttrRegexps` in this PR (there's enough new logic it in already :tired_face: ), but we could probably generate it at runtime down the road in the registry [here](https://github.com/plotly/plotly.js/blob/143749ecfecf676adfaa42ff03469e273c3f8cea/src/registry.js#L92-L95)?

Isn't `_arrayAttrRegexps` simply,

```js
allIsLinkedToArrayAttributes.map((attr) => {
  return module.name + '\[[(0|[1-9][0-9]+)\]\.' + attr;
});
```